src/abstractinterpret/inferenceerrorreport.jl

```diff
@@ -94,21 +95,21 @@ end
 # signature
 # ---------
 
-@inline get_sig(s::StateAtPC, @nospecialize(x=get_stmt(s))) = Signature(get_sig_nowrap(s, x))
+@inline get_sig(s::StateAtPC, @nospecialize(x=get_stmt(s))) = Signature(get_sig_nowrap(s, x)...)
 get_sig(sv::InferenceState) = get_sig((sv, get_currpc(sv)))
 
-get_sig(mi::MethodInstance) = Signature(Any[mi])
+get_sig(mi::MethodInstance) = Signature(Any[mi], mi.specTypes)
 get_sig(caller::InferenceResult) = get_sig(get_linfo(caller))
 
 function get_sig_nowrap(@nospecialize args...)
     sig = Any[]
-    handle_sig!(sig, args...)
-    return sig
+    sig, tt = handle_sig!(sig, args...)
```

Comment:
I was hoping there is a good way to compute `tt`, but I couldn't come up with anything immediately. It’s likely that by making some adjustments to Julia base's abstract interpretation routines, we could obtain more solid information. Leave that extension to me.